models/sdn-fabric-0.1.x/yang/onf-switch-port@2022-05-25.yang

```diff
@@ -52,12 +52,10 @@ submodule onf-switch-port {
       uses st:desc-display-name;
 
       leaf speed {
-      /** TODO fix when handling of leaf-list of enum is fixed in YangNodeNavigator
-       * must "set-contains(/switch-model[@switch-model-id=$this/../../model-id]/port/speeds, .)" {
-       *   error-message "port speed must be present in corresponding switch-model/port";
-       *   description "port speed must be from the corresponding switch-model";
-       * }
-       */
+        must "contains(/switch-model[@switch-model-id=$this/../../model-id]/port/speeds, string($this))" {
```

Comment:
@frouzbeh this is the important statement here - this is a YANG must statement that adds additional rules in the YANG model - we call these Guard Rails

See [https://datatracker.ietf.org/doc/html/rfc6020#section-7.5.3](https://datatracker.ietf.org/doc/html/rfc6020#section-7.5.3)

Reply:
I see, thanks.